File: src/main/java/io/koju/autopos/catalog/service/ItemService.java
package io.koju.autopos.catalog.service;

import io.koju.autopos.catalog.domain.Item;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ItemService {

    Item save(Item item);

    Page<Item> findAll(String queryTerm, Pageable pageable);

    Item findOne(Long id);

    void delete(Long id);

    void adjustQuantity(Item item, Integer number);

    default void addQuantity(Item item, Integer number) {
        adjustQuantity(item, number);
    }

    default void subtractQuantity(Item item, Integer number) {
        adjustQuantity(item, -number);
    }

}

File: src/main/scala/io/koju/autopos/catalog/service/ItemService.scala
package io.koju.autopos.catalog.service

import java.lang.Long

import io.koju.autopos.catalog.domain.Item
import org.springframework.data.domain.{Page, Pageable}

trait ItemService {

  def save(item: Item): Item

  def findAll(queryTerm: String, pageable: Pageable): Page[Item]

  def findOne(id: Long): Item

  def delete(id: Long)

  def adjustQuantity(item: Item, number: Integer)

  def addQuantity(item: Item, number: Integer) {
    adjustQuantity(item, number)
  }

  def subtractQuantity(item: Item, number: Integer) {
    adjustQuantity(item, -number)
  }
}